cql/ManageCommonAbnormality.cql

```diff
@@ -365,8 +372,84 @@ define PositivePrimaryHpv:
 define MostRecentCytologyCotestWasAfterAge21:
   Collate.MostRecentCytologyCotest.date >= Patient.birthDate + 21 years
 
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+// Guidance for expedited Treatment (rarely screened) (Box 1. Essential Changes from Prior Management Guidelines, #3)
+// Expedited treatment is preferred for never or rarely screened patients with HPV-positive HSIL cytology regardless of HPV genotype.
+
+define RarelyScreenedLookbackInterval:
+  Interval[Collate.MostRecentCytologyCotest.date - Collate.RarelyScreenedInterval, Collate.MostRecentCytologyCotest.date)  
+  
+
+define IsHsilAndHpvPositive:
+  Collate.MostRecentCytologyCotest is not null and
+  Collate.MostRecentCytologyCotestResult = 'HSIL+' and
+  HighRiskHPVPositiveResults
+  
+define HighRiskHPVPositiveResults:
+  AnyTrue(
+    (Collate.MostRecentHpvReport.allConclusions) aC 
+      return aC in Dash."High Risk HPV Positive Results"
+  )
+  
+define PreviousCytologyOrSurveillanceIsOldOrAbsent:
+  (
+    Collate.SecondMostRecentCytologyReport is null and
+    Collate.SecondMostRecentCytologyCotest is null and
+    Collate.SecondMostRecentHpvReport is null
+  ) or
+  (
+    Collate.SecondMostRecentCytologyReport is not null and
+    not (Collate.SecondMostRecentCytologyReport.date included in RarelyScreenedLookbackInterval)
+  ) or
+  (
+    Collate.SecondMostRecentCytologyCotest is not null and 
+    not (Collate.SecondMostRecentCytologyCotest.date included in RarelyScreenedLookbackInterval)
+  ) or
+  ( 
+    Collate.SecondMostRecentHpvReport is not null and
+    not (Collate.SecondMostRecentHpvReport.date included in RarelyScreenedLookbackInterval)
+  )
+
+define IsRarelyScreenedPatient: 
+  AgeInYears() >= 27 and 
```

Comment:
Why does this start at 27 and not 25?

I know this was discussed at some point, but I'm not sure if this was over email, Jira, GH Issues, or elsewhere. Is it because a patient isn't supposed to start screening until 21, so we want to give them 6 years (5 years + 1 year wiggle room) before considering them rarely screened? If that is the case, why does it matter what the patient was _supposed_ to be doing when we're calculating rarely screened?

Reply:
This was discussed over email with Ginny. She decided on 27. See email from 8/15 re: Rarely Screened.

Reply:
I don't think this is up to us to decide. Doesn't setting an age limit of 27 years old here directly conflict with the main guideline article, which states:

>  For non-pregnant patients 25 years or older, expedited treatment, defined as treatment without preceding colposcopic biopsy demonstrating CIN 2+, is preferred when the immediate risk of CIN 3+ is ≥60%, and is acceptable for those with risks between 25% and 60%. Expedited treatment is preferred for nonpregnant patients 25 years or older with high-grade squamous intraepithelial lesion (HSIL) cytology and concurrent positive testing for HPV genotype 16 (HPV 16) (i.e., HPV 16–positive HSIL cytology) and never or rarely screened patients with HPV-positive HSIL cytology regardless of HPV genotype.

I still think this is far over-complicating the definition of rarely screened. It would be nice for rarely screened to be an objective state: either they were screened in the last 5.5 years, or they were not. I don't understand why a 25 year old would not be considered rarely screened, simply because they were not required to be screened 5 years ago. They still haven't been screened in the last 5 years, bottom line. And more importantly, I do not see where in the guidelines it mentions that a patient cannot be considered rarely screened until they are 27.